esphome/components/online_image/png_image.cpp

```diff
@@ -57,6 +59,7 @@ size_t HOT PngDecoder::decode(HTTPClient &http, WiFiClient *stream, std::vector<
     }
 
     int len = stream->readBytes(buffer + remain, size);
+    pending -= len;
```

Comment:
instead of recalculating "pending", it could be renamed to total_size and stay constant; and then compare it against "total"; which is already tracking how much has already been downloaded.